code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -1067,3 +1115,150 @@ Pass a positive integer as an argument to override a bot's default speed.
 
 /mob/living/simple_animal/bot/rust_heretic_act()
 	adjustBruteLoss(400)
+
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+//Multi-Z Related section for NSV13
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/**
+ * Finds nearest bot elevator.
+ *
+ * Arguments:
+ * * direciton - UP or DOWN.
+ */
+/mob/living/simple_animal/bot/proc/find_nearest_bot_elevator(direction)
+	if(!direction)
+		return
+	if(direction != UP && direction != DOWN)
+		return
+
+	var/target
+	for(var/obj/structure/bot_elevator/elevat in GLOB.bot_elevator)
+		if(elevat.z != z)
+			continue
+		if(direction == UP && !elevat.up)
+			continue
+		if(direction == DOWN && !elevat.down)
+			continue
+		if(!target)
+			target = elevat
+			continue
+		if(get_dist_euclidian(elevat, src) > get_dist_euclidian(target, src))
+			continue
+		target = elevat
+	return target
+
+/**
+ *
+ * Makes the bot move up or down a Z-level depending on the bot_z_mode
+ * and the original destination
+ */
+/mob/living/simple_animal/bot/proc/bot_z_movement()
+	var/obj/structure/bot_elevator/E = locate(/obj/structure/bot_elevator) in get_turf(src)
+	if(bot_z_mode == 10)
+		if(E)
+			if(z > last_waypoint.z)
+				E.travel(FALSE, src, FALSE, E.down, FALSE)
+				ai_waypoint = last_waypoint
+				call_bot(calling_ai, ai_waypoint)
+			else
+				E.travel(TRUE, src, FALSE, E.up, FALSE)
+				ai_waypoint = last_waypoint
+				call_bot(calling_ai, ai_waypoint)
+
+	if(bot_z_mode == 20)
+		if(E)
+			if(z > original_patrol.z)
+				E.travel(FALSE, src, FALSE, E.down, FALSE)
+				patrol_target = original_patrol
+				calc_path()
+			else
+				E.travel(TRUE, src, FALSE, E.up, FALSE)
+				patrol_target = original_patrol
+				calc_path()
+	if(bot_z_mode == 30)
+		if(E)
+			if(z > last_summon.z)
+				E.travel(FALSE, src, FALSE, E.down, FALSE)
+				summon_target = last_summon
+				calc_summon_path()
+			else if(z < last_summon.z)
+				E.travel(TRUE, src, FALSE, E.up, FALSE)
+				summon_target = last_summon
+				calc_summon_path()
+
+//BOT MULTI-Z MOVEMENT
+/mob/living/simple_animal/bot/proc/call_bot_z_move(caller, turf/ori_dest, message=TRUE)
+	//For giving the bot temporary all-access.
+	var/obj/item/card/id/all_access = new /obj/item/card/id
+	var/datum/job/captain/All = new/datum/job/captain
```

Comment:
weird capitalized var name

Reply:
var/datum/job/captain/All is a direct copy from another section of this code, it's something that already exists in the bot code even before I started modifying it.